### ISSUE TITLE
[SALTSTACK] -Unable to install sift

### ISSUE THREAD

**Issue by Cybertest06** [teamdfir/sift#581]
Getting error as below
sudo sift install
> sift-cli@1.11.0+0-g966729b
> sift-version: notinstalled

> mode: desktop
Installing and configuring SaltStack properly ...

Command failed: wget -O - https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3001/salt-archive-keyring.gpg | apt-key add -
--2022-11-04 17:24:11--  https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3001/salt-archive-keyring.gpg
Resolving repo.saltproject.io (repo.saltproject.io)... 52.84.45.4, 52.84.45.104, 52.84.45.63, ...
Connecting to repo.saltproject.io (repo.saltproject.io)|52.84.45.4|:443... Warning: apt-key output should not be parsed (stdout is not a terminal)
connected.
HTTP request sent, awaiting response... 404 Not Found
2022-11-04 17:24:12 ERROR 404: Not Found.

gpg: no valid OpenPGP data found.

Error: Command failed: wget -O - https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3001/salt-archive-keyring.gpg | apt-key add -
--2022-11-04 17:24:11--  https://repo.saltproject.io/py3/ubuntu/20.04/amd64/3001/salt-archive-keyring.gpg
Resolving repo.saltproject.io (repo.saltproject.io)... 52.84.45.4, 52.84.45.104, 52.84.45.63, ...
Connecting to repo.saltproject.io (repo.saltproject.io)|52.84.45.4|:443... Warning: apt-key output should not be parsed (stdout is not a terminal)
connected.
HTTP request sent, awaiting response... 404 Not Found
2022-11-04 17:24:12 ERROR 404: Not Found.

gpg: no valid OpenPGP data found.

    at ChildProcess.exithandler (child_process.js:303:12)
    at ChildProcess.emit (events.js:315:20)
    at maybeClose (internal/child_process.js:1051:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)

----- PLEASE READ ----------------------

A lot of failures are caused by the apt system being locked
or unhealthy. 

Before opening an issue in GitHub, please check to see if 
your apt system is healthy.

Try running 'apt-get update' then remove any packages that 
aren't used by running 'apt-get autoremove'

**Comment by jeremyng123** [comment]
> @digitalsleuth I believe updating the SIFT CLI will not work.
> 
> this error occurs right now because there is an update in one of the dependencies that saltstack requires (`importlib-metadata`).
> 
> This library was updated on Oct 2 2022. This breaks the install process because `importlib-metadata` has changed the way `entry_point` datatype was used (previously `dict` type, now its a `list` type) Similar issues were found by other users that depends on this library [Breaking change in 7e5bae4 (importlib_metadata 5)](https://github.com/python/importlib_metadata/issues/409)
> 
> A simple fix would be to **downgrade** `importlib-metadata` python module (latest: 5.0.0) to the previous version (4.13.0).
> 
> I have done this change myself and the installation has since run smoothly.

Found that the solution was already mentioned in another issue: #579

**Comment by ekristen** [comment]
Closing due to age, this has been addressed in the latest version of the SIFT CLI. Please also note, that once https://github.com/teamdfir/sift-saltstack/pull/80 is merged, SIFT CLI will be officially deprecated as of March 1, 2023, and is replaced with [cast](https://github.com/ekristen/cast)
